lib/active_model/serializer/adapter/json/fragment_cache.rb

```diff
@@ -1,16 +1,6 @@
-require 'active_model/serializer/adapter/fragment_cache'
-module ActiveModel
-  class Serializer
-    class Adapter
-      class Json < Adapter
-        class FragmentCache
+class ActiveModel::Serializer::Adapter::Json::FragmentCache
```

Comment:
Great discussion, indeed I agree that the previous class syntax was more readable, but have the hook might be a good trade off, this is a big refactor and also add some new behaviors, let's keep it the way it is now, and try to address a way of doing this and gaining back readability on other PR.